src/components/sbb-link/sbb-link.tsx

```diff
@@ -25,195 +25,154 @@ import { hostContext } from '../../global/helpers/host-context';
 
-  @Element() public el!: HTMLElement;
+  /**
+   * This will be forwarded as aria-describedby to the relevant nested element.
+   */
+  @Prop() public accessibilityDescribedby: string | undefined;
+
+  /**
+   * This will be forwarded as aria-labelledby to the relevant nested element.
+   */
+  @Prop() public accessibilityLabelledby: string | undefined;
 
   public connectedCallback(): void {
     // Check if the current element is nested in either an `<a>` or `<button>` element.
     this._isStatic = !!hostContext('a,button', this.el);
   }
 
   /**
-   * Get the attributelist base on the config and the resulting element
-   * @return <object>
-   */
-  private _getAttributeList(): object {
-    const currentLanguage = getDocumentLang();
-    const currentWritingMode = getDocumentWritingMode();
-    const attributeList: Record<string, string> = {};
-
-    Object.assign(attributeList, {
-      dir: currentWritingMode,
-      class: this._getClassString(),
-      'aria-label': this.accessibilityLabel || undefined,
-      'aria-labelledby': this.accessibilityLabelledby || undefined,
-      'aria-describedby': this.accessibilityDescribedby || undefined,
-      name: this.name || undefined,
-      id: this.idValue || undefined,
-    });
-
-    const openInNewWindow = !window.location.href.includes(this.href);
-    if (openInNewWindow && !this._isStatic && this.href) {
-      Object.assign(attributeList, {
-        rel: 'external noopener nofollow',
-        target: '_blank',
-        'aria-label':
-          `${attributeList['aria-label']}. ${i18nTargetOpensInNewWindow[currentLanguage]}` ||
-          undefined,
-      });
-    }
-
-    if (this._isStatic) {
-      return attributeList;
-    } else if (this.href) {
-      // Anchor case
-      return Object.assign(attributeList, {
-        href: this.href,
-        download: this.download ? '' : undefined,
-        tabIndex: this.disabled ? '-1' : undefined,
-      });
-    }
-
-    // Button case
-    return Object.assign(attributeList, {
-      type: this.type || undefined,
-      form: this.form || undefined,
-      disabled: this.disabled ? 'true' : undefined,
-      onClick: this._emitButtonClick.bind(this),
-    });
-  }
-
-  private _emitButtonClick(): void {
+   * Method triggered at button click to emit the click event (can be caught from parent component).
+   */
+  public emitButtonClick(): void {
     if (!this.disabled) {
       this.click.emit(this.eventId);
     }
   }
 
+  /**
+   * Generate the class attribute based on component's parameters.
+   */
   private _getClassString(): string {
     const textSizeClass = this.variant === 'inline' ? '' : ` sbb-link--text-${this.textSize}`;
     const iconPositionClass =
-      this.iconPlacement === 'start'
-        ? ' sbb-link--icon-placement-start'
-        : ' sbb-link--icon-placement-end';
+      this.iconPlacement === 'end' ? ` sbb-link--icon-placement-${this.iconPlacement}` : '';
```

Comment:
At this point we could further refactor the property by making it as a boolean: "placeIconOnEnd" (?)

Reply:
Done in commit 9f3ac8a572b2aa5cc8642f308da86c315302837d